code/game/objects/structures/headpike.dm

```diff
@@ -0,0 +1,51 @@
+/obj/structure/headpike
+	name = "spooky head on a spear"
+	desc = "When you really want to send a message."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "headpike"
+	density = FALSE
+	anchored = TRUE
+
+/obj/structure/headpike/bone //for bone spears
+	icon_state = "headpike-bone"
+
+
+/obj/structure/headpike/CheckParts(list/parts_list)
+	..()
+	var/obj/item/bodypart/head/H = locate() in contents
+	if(H)
+		update_icon()
+		name = "[H.real_name]'s head on a spear"
+
+/obj/structure/headpike/Initialize()
+	. = ..()
+	pixel_x = rand(-8, 8)
+
+/obj/structure/headpike/update_icon()
+	..()
+	var/obj/item/bodypart/head/H = locate() in contents
+	var/mutable_appearance/MA = new()
+	if(H)
+		MA.overlays = H.overlays.Copy()
```

Comment:
It pains me to delay this PR, but you'll have to wait for #28233 to be merged so you can do this the right way